cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/DatastoreImplForceInsertTest.java

```diff
@@ -505,6 +505,30 @@ public void forceInsert_pickWinnerOfConflicts_Simple_2y_last() throws Exception
     }
 
     @Test
+    public void forceInsert_pickWinnerOfConflicts_TwoOfSameRoot_OneDeleted() throws Exception {
+        DocumentRevision rev1a = createDbObject("1-x", bodyOne);
+        DocumentRevision rev1b = createDbObject("1-x", bodyTwo);
+        DocumentRevision rev2 = createDbObjectDeleted("2-x");
+
+        // Insert the same document twice, so we have two identical roots
+        // We have given them different bodies so we can tell them apart
+        datastore.forceInsert(rev1a, "1-x");
+        datastore.forceInsert(rev1b, "1-x");
+
+        // Now force insert the deleted
+        datastore.forceInsert(rev2, "1-x", "2-x");
+
+        // Get the document
+        DocumentRevision doc = datastore.getDocument(OBJECT_ID);
+
+        // We favour non-deleted nodes so the winner should be a 1-x
+        Assert.assertEquals("The document winner should be the expected revision", "1-x", doc
+                .getRevision());
+        Assert.assertEquals("The document winner should have the expected body", bodyTwo.asMap(),
```

Comment:
There is no `order by` clause in the SQL used for `pickWinnerOfConflicts` so by my reading of the code it is arbitrary which sequence is last put into the map for a given rev id. Possibly the underlying SQLite structure is sorted in the order in which rows were inserted but we can't rely on that.

Not that it really matters, if we have got into the situation where two documents have exactly the same revision ID then we have no way of distinguishing which one is "right" anyway. You could be consistent and pick the lowest/highest sequence number on the basis that first/last write wins.